core/libdeno.rs

```diff
@@ -220,112 +994,1206 @@ type deno_resolve_cb = unsafe extern "C" fn(
+        (mod_id != 0 && error_str == nullptr) ||
+        (mod_id == 0 && !d->last_exception_handle_.IsEmpty()));
+  deno::UserDataScope user_data_scope(d, user_data);
+
+  auto* isolate = d->isolate_;
+  v8::Isolate::Scope isolate_scope(isolate);
+  v8::Locker locker(isolate);
+  v8::HandleScope handle_scope(isolate);
+  auto context = d->context_.Get(d->isolate_);
+  v8::Context::Scope context_scope(context);
+
+  */
+
+  let deno_isolate: &mut DenoIsolate = unsafe { std::mem::transmute(i) };
+  assert!(
+    (mod_id == 0 && error_str.is_some())
+      || (mod_id != 0 && error_str.is_none())
+      || (mod_id == 0 && !deno_isolate.last_exception_handle_.is_empty())
+  );
+
+  let user_data: *mut c_void = unsafe { std::mem::transmute(user_data) };
+  let user_scope = UserDataScope::new(deno_isolate, user_data);
+
+  let isolate = deno_isolate.isolate_.as_ref().unwrap();
+  let mut locker = v8::Locker::new(isolate);
+  let mut hs = v8::HandleScope::new(&mut locker);
+  let scope = hs.enter();
+  assert!(!deno_isolate.context_.is_empty());
+  let mut context = deno_isolate.context_.get(scope).unwrap();
+  context.enter();
+
+  /*
+  auto it = d->dyn_import_map_.find(import_id);
+  if (it == d->dyn_import_map_.end()) {
+    CHECK(false);  // TODO(ry) error on bad import_id.
+    return;
+  }
+
+  /// Resolve.
+  auto persistent_promise = &it->second;
+  auto promise = persistent_promise->Get(isolate);
+
+  auto* info = d->GetModuleInfo(mod_id);
+
+  // Do the following callback into JS?? Is user_data_scope needed?
+  persistent_promise->Reset();
+  d->dyn_import_map_.erase(it);
+
+  if (info == nullptr) {
+    // Resolution error.
+    if (error_str != nullptr) {
+      auto msg = deno::v8_str(error_str);
+      auto exception = v8::Exception::TypeError(msg);
+      promise->Reject(context, exception).ToChecked();
+    } else {
+      auto e = d->last_exception_handle_.Get(isolate);
+      ClearException(context);
+      promise->Reject(context, e).ToChecked();
+    }
+  } else {
+    // Resolution success
+    Local<Module> module = info->handle.Get(isolate);
+    CHECK_EQ(module->GetStatus(), v8::Module::kEvaluated);
+    Local<Value> module_namespace = module->GetModuleNamespace();
+    promise->Resolve(context, module_namespace).ToChecked();
+  }
+  d->isolate_->RunMicrotasks();
+  */
+
+  let mut resolver_handle = match deno_isolate.dyn_import_map_.remove(&id) {
+    Some(handle) => handle,
+    None => {
+      // TODO(ry) error on bad import_id.
+      panic!()
+    }
+  };
+
+  /// Resolve.
+  let mut resolver = resolver_handle.get(scope).unwrap();
+  resolver_handle.reset(scope);
+
+  let maybe_info = deno_isolate.get_module_info(mod_id);
+
+  if maybe_info.is_none() {
+    // Resolution error.
+    if error_str.is_some() {
+      let msg = v8::String::new(scope, &error_str.unwrap()).unwrap();
+      // FIXME: need to enter isolate here
+      let e = v8::type_error(scope, msg);
+      resolver.reject(context, e).unwrap();
```

Comment:
@ry errors use `GetCurrent()` API so we need to call `isolate.enter()` before creating the error

Reply:
cc @piscisaureus